### ISSUE TITLE
test code running give unexpected results

### ISSUE THREAD

**Issue by fuxianh** [ShangtongZhang/DeepRL#68]
def test(opt):
    torch.manual_seed(1)
    task =  Task(opt.g)
    model = torch.load( './{}/{}/20000000.pth'.format(opt.saved_path ,opt.g))
    model = model.cuda()
    model.eval()
    model = model.double()
    init_s = task.reset()
    state = torch.from_numpy(np.asarray(init_s))
    done = True
    reward_episode = 0
    episode_len = 0
    episode_num = 0
    while True:
        # state = state*1.0/255
        q = model(state.type(dtype=torch.DoubleTensor).cuda())
        # action = np.random.randint(0,4)
        action = to_np(q.argmax(-1))
        print('q value: {}'.format(q))
        print('action: {}'.format(action))
        # q_dist = to_np(q)

        state, reward, done, info = task.step([action])
        state = torch.from_numpy(np.asarray(state))
        reward_episode += reward[0]
        episode_len += 1
        if done:
            episode_num += 1
            reward_episode = 0
            episode_len = 0
            init_s = task.reset()
            state = torch.from_numpy(np.asarray(init_s))

**Comment by fuxianh** [comment]
Hi Shangtong, I write a test code shown in the above box,  the model saved should be fine and done as : torch.save(agent.network, './model/{}/{}.pth'.format( config.game, agent.total_steps)) .
The problem is, every time I test, the action selected by argmax q value tends to be the same one. But the corresponding training result is very good when save that model. Could you please help me solve this problem? Thank you very much.

**Comment by ShangtongZhang** [comment]
Sorry I cannot help issues like this.
